packages/material-ui-lab/src/Masonry/Masonry.js

```diff
@@ -81,48 +106,84 @@ const Masonry = React.forwardRef(function Masonry(inProps, ref) {
   });
 
   const masonryRef = React.useRef();
+  const [maxColumnHeight, setMaxColumnHeight] = React.useState();
+  const [maxNumOfRows, setMaxNumOfRows] = React.useState();
   const { children, className, component = 'div', columns = 4, spacing = 1, ...other } = props;
-  const ownerState = { ...props, spacing, columns };
+  const ownerState = { ...props, spacing, columns, maxColumnHeight, maxNumOfRows };
   const classes = useUtilityClasses(ownerState);
+  const lineBreakStyle = {
+    flexBasis: '100%',
+    width: 0,
+    margin: 0,
+    padding: 0,
+    content: '',
+  };
 
-  const contextValue = React.useMemo(() => ({ spacing }), [spacing]);
-  let didWarn = false;
   React.useEffect(() => {
-    // scroller always appears when masonry's height goes beyond 2,000px on Chrome
-    const handleScroll = () => {
-      if (masonryRef.current.clientHeight === 1998 && !didWarn) {
-        console.warn(
-          [
-            'Material-UI: The Masonry can have the maximum height of 2,000px on Chrome browser.',
-            'Items that go beyond this height fail to be rendered on Chrome browser.',
-            'You can find more in this open issue: https://github.com/mui-org/material-ui/issues/27934',
-          ].join('\n'),
-        );
-
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-        didWarn = true;
+    const handleResize = () => {
+      const columnHeights = [];
+      const numOfRows = [];
+      let skip = false;
+      Array.from(masonryRef.current.children).forEach((child) => {
+        if (child.className.includes('line-break') || skip) {
```

Comment:
This fails on every `class` attribute that contains `line-break` but not the actual classname you hard-coded below e.g. `class="my-line-break". 

Do we really need a classname or can we leverage a custom data attribute instead? If we need a classname then 
```suggestion
        if (child.classList.contains("line-break") || skip) {
```
should be used instead.

Reply:
Thanks! It is necessary to neglect line breaks in computing heights. I replaced `line-break` with `classses.break` in this commit: https://github.com/mui-org/material-ui/pull/28059/commits/ecb00232564767c05528060fa2135af22ae85caf

Reply:
In fact, using data-attribute is enough and much simpler. I made the change in this commit: